### ISSUE TITLE
Status bar button occupies quite some space

### ISSUE THREAD

**Issue by bartosz-antosik** [James-Yu/LaTeX-Workshop#29]
Could you maybe please make the status bar button "Compile-on-save Enabled" make (optionally) smaller, e.g. without the text that accompanies the switch?

It occupies quite some space, having it reduced would make it less obtrusive, as it persists no mater what file type one is working with.

**Comment by James-Yu** [comment]
Do you think the button is necessary? I am thinking of ways to make it intuitive and short but yet to find one. Any suggestions are greatly appreciated.

**Comment by Inachos** [comment]
I think the button may have the general "what if all plugins did this" problem. Is it possible to to a line in the right-click menu that either says enable or disable depending on the state of compileonsave?

**Comment by James-Yu** [comment]
I agree with you @Inachos. Will see if there's method to move it to context menu.

**Comment by Inachos** [comment]
@James-Yu Another two cents worth: the Python extension, which is maintained by a Microsoft employee, has settings for linting and autoformatting on save. And they only put these settings in the user-settings json, and nowhere else (no button or context menu). 

I do agree the context menu isn't a bad place to put it, but if it's not working out, common practice seems to be to pick the default that people will most likely want, and let them change it via the settings otherwise.

(for example, they enabled linting, but disabled formatting by default)

**Comment by bartosz-antosik** [comment]
Let me too!

Visual control over state is (sometimes) a good thing, when command is of flip type (ON->OFF->ON...) But It could be optional. And smaller, less obtrusive, please. Like two tone PDF doc icon only? Like light gray for disabled & white for enabled? To those who know what to expect it says it all.

I would love to see ON & OFF among extension's commands called with Ctrl+P list. It may be convenient (e.g. with large documents) to at times just save the document otherwise update it in preview.

Also I totally agree that having system-wide default setting and workspace setting works the best (this is how I understand @Inachos post above).

Summing up: Settable from settings to some desired default state (global OR global + workspace), commands to set ON and OFF respectively "on the fly" & optional icon.

I feel like I am maybe messing up a bit. Sorry.

**Comment by James-Yu** [comment]
I come up with an alternative. Now the statusbar item will collapse to a single item after 2 seconds, which should occupy only a little bit of space. In addition, the feature can be controlled by both global and workspace settings. Last but not least, it can always be toggled by `Toggle LaTeX Compile-on-save` command.

Is that a solution? Discussion welcomed!

**Comment by James-Yu** [comment]
Add screenshots on the two states.

![screenshot from 2017-03-17 14 29 27](https://cloud.githubusercontent.com/assets/4210342/24031556/375286ec-0b1e-11e7-9a3b-18484ccc4380.png)
![screenshot from 2017-03-17 14 28 34](https://cloud.githubusercontent.com/assets/4210342/24031557/375552c8-0b1e-11e7-88b2-b5edd520f38f.png)

**Comment by bartosz-antosik** [comment]
Looks very ok. Collapsing as described seems to me a good idea as I read about it.

There's one more thing I considered in the meantime - would be excellent if the status bar item would only be present for 'latex' type documents.

Do I understand correctly that it is not yet pushed to production so the extension does not update in VSCode?

**Comment by Inachos** [comment]
Thank you so much. Next week I will do a clone and see if i can get into the code. Wisecracking is so much easier than contributing.

**Comment by James-Yu** [comment]
No it isn't, as I would like to hear for your comments before shut the door. Will add the latex only feature and release the update. Soon(TM)

**Comment by James-Yu** [comment]
version `0.2.20` is now live and have all the changes. Have a try!
